server/src/modules/clubs/infra/repos/implementations/mikro-club-repo.ts

```diff
@@ -0,0 +1,35 @@
+import { QueryOrder } from '@mikro-orm/core' // {EntityRepository, QueryOrder}
+import { DB } from '../../../../../shared/infra/db'
+import { ClubEntity } from '../../../../../shared/infra/db/entities/club.entity'
+import { Club } from '../../../domain/entities/club'
+import { ClubMap } from '../../../mappers/club-map'
+import { ClubRepo } from '../club-repo'
+
+export class MikroClubRepo implements ClubRepo {
+  // TODO: same coupling error as equivalent users version of this file
+
+  // private clubsEntityRepo: EntityRepository<ClubEntity> | undefined
+
+  // constructor(clubRepo: EntityRepository<ClubEntity> | undefined) {
+  //   this.clubsEntityRepo = clubRepo
+  // }
+
+  async exists(name: string): Promise<boolean> {
+    const club = await DB.clubsEntityRepo.findOne({ name })
+    return club !== null
+  }
+
+  async getClubByClubId(clubId: string): Promise<Club> {
+    const club = await DB.clubsEntityRepo.findOne({ id: clubId })
+    if (!club) throw new Error() // TODO handle error
```

Comment:
i'm leaning towards yes, but it would be nice to have async/await support with them. i think we should look into using a production ready package now